packages/react-next/src/components/Checkbox/Checkbox.types.ts

```diff
@@ -129,10 +130,17 @@ export interface ICheckboxProps extends React.ButtonHTMLAttributes<HTMLElement |
   checkmarkIconProps?: IIconProps;
 
   /**
-   * Optional keytip for this checkbox
+   * Optional keytip.
+   *
+   * @deprecated Use keytipData instead.
    */
   keytipProps?: IKeytipProps;
 
+  /**
+   * Optional keytip data.
+   */
+  keytipData?: IKeytipData;
```

Comment:
Are we ok with still having `keytip` related props in the components?

Reply:
the question here is - how can we give the keytip data to a component. one alternative way to do this is by creating a composed component. but `compose` won't be ready for all the components. the way i'm approaching (and the migration story) is applicable to all functional components we converted in V8

Reply:
this doesn't seem correct.

Keytips should be designed to be able to work with any component regardless of them knowing about keytip. The requirement is that the keytip is able to identify which element to target the tooltip towards, and which element to click when keytip is executed.

In composed components, that may mean you pass ids into the particular slots. Or you add refs. But I think we need a model that doesn't require keytipData to be on every component in the world.

Reply:
@dzearing  - i do agree having the `keytipData` pollutes prop interface. here is what i was thinking - the way to achieve not having `keytipData` prop is via `compose`. `compose` will allow user to pass the attributes down to any slot they want. however before we can announce `compose` to user to use, having `keytipData` prop is a nice stepping stone. 

On what you proposed:
- if we define IDs within our component, how do we communicate these "IDs" from our component to Keytip users so they know what ID to set?
- other than target/execute element ids, we also need to set `aria-describedby` within the targeted component. the `aria-describedby` is ID of element which created by `KeytipLayer` that describes the keytip sequences
![image](https://user-images.githubusercontent.com/1207059/85903721-e65f6400-b7bb-11ea-9803-e62ac036cb0b.png)


![image](https://user-images.githubusercontent.com/1207059/85903780-0a22aa00-b7bc-11ea-8e00-6a5dd0ec8514.png)